### PR TITLE
Backgrounds Explained, aka, "a background on backgrounds"

### PR DESCRIPTION
**Pull request by Lokathor**

### CODE REVIEW

File: book/src/ch03/regular_backgrounds.md [hunk 1]
```diff
@@ -1,2 +1,301 @@
...
+Now, at either 256 or 512 tiles per charblock, you might be thinking that with a
+10 bit index you can index past the end of one charblock and into the next.
+You'd be right, mostly.
+
+As long as you stay within the background memory region for charblocks (that is,
+0 through 3), then it all works out. However, if you try to get the background
+rendering to reach outside of the background charblocks you'll get an
+implementation defined result. It's not the dreaded "undefined behavior" we're
+often worried about in programming, but the results _are_ determined by what
+you're running the game on. With real hardware, you get a bizarre result
```
Comment (ketsuban) [inline book/src/ch03/regular_backgrounds.md]: Well, on the GBA. On the DS you just get zeroes (i.e. blank, transparent tiles).

Comment (Lokathor) [reply book/src/ch03/regular_backgrounds.md]: Let's hold that off until we do the follow up book: rust on the DS!

Comment (ketsuban) [reply book/src/ch03/regular_backgrounds.md]: As in on the DS *in GBA mode*. You can't even rely on it being a bizarre result.

Comment (Lokathor) [reply book/src/ch03/regular_backgrounds.md]: Ah, well i can add that. Seems i missed at least a few changes in the last push so i have to do at least one more for this PR

**Pull request merged**